trieste/acquisition/function.py

```diff
@@ -275,29 +273,23 @@ def __init__(
             minimum, else use Gumbel sampling.
         :param num_fourier_features: Number of Fourier features used for approximate Thompson
             sampling. If None, then do exact Thompson sampling.
+        :raise tf.errors.InvalidArgumentError: If
+
+            - ``num_samples`` or ``grid_size`` are negative, or if
+            - ``num_fourier_features`` is negative or zero
+            - ``num_fourier_features`` is specified an ``use_thompson`` is `False`
         """
-        self._search_space = search_space
+        tf.debugging.assert_positive(num_samples)
+        tf.debugging.assert_positive(grid_size)
 
-        if num_samples <= 0:
-            raise ValueError(f"num_samples must be positive, got {num_samples}")
-        self._num_samples = num_samples
+        if num_fourier_features is not None:
+            tf.debugging.Assert(use_thompson, [])
```

Comment:
i've dropped err messages in a few places because `Assert` doesn't have a message argument

Reply:
i could perhaps put them back in as separate `tf.print` statements

Reply:
strange it doesn't have message argument, but `tf.print` probably isnt a good solution either, lets leave it as is